modules/govuk_search/manifests/gor.pp

```diff
@@ -0,0 +1,48 @@
+# == Class: govuk_search::gor
+#
+# Dump POST traffic to file so that it can be replayed. This means we have a simple
+# way to recover from failure as we can restore the last backup and then replay the
+# traffic (POST requests are for data insert/update only)
+#
+# Traffic can then be replayed by follow the guide at:
+# https://github.com/buger/goreplay/wiki/Capturing-and-replaying-traffic
+#
+# === Parameters
+#
+# [*enabled*]
+#   Boolean to determine if Search traffic will be saved; defaults to false.
+#
+# [*output_path*]
+#   File location for files to be saved to.
+#
+class govuk_search::gor (
+  $output_path = undef,
+  $enabled = false
+) {
+
+  validate_bool($enabled)
+
+  if($enabled) {
+    validate_re($output_file, '/.*')
```

Comment:
@MatMoore added some tests and think this might now work, will test when I'm in the office tomorrow